caffe2/contrib/gloo/allreduce_ops.h

```diff
@@ -65,12 +65,12 @@ class AllreduceOp final : public Operator<Context> {
 
     // Verify inputs == outputs
     CAFFE_ENFORCE_EQ(init_.inputs.size(), init_.outputs.size());
-    for (auto i = 0; i < init_.inputs.size(); i++) {
+    for (auto i = 0U; i < init_.inputs.size(); i++) {
```

Comment:
Personally I prefer for (size_t i = 0; ...), but this is fine.

Reply:
yeah I think size_t is better, I'll change it after my stack of diffs land

Reply:
I mean I'll change this in my code

Reply:
Please loop @zdevito on that one, because I had `size_t` there initially and he suggested to use `unsigned int` out of performance considerations.

Reply:
But personally I agree, that if upper limit it 64-bit unsigned value, then loop iterator should also be 64-bit unsigned value.

Reply:
Actually all of params.inputs, devices_ or elements are std::vectors. While in std::vector the return type of std::vector<T>::size() is std::vector<T>::size_type, which in most cases is size_t. If there is platform concerns, I think we'd better use std::vector<T>::size_type i = 0; And I don't think there should be performance issues for such a for-loop iterative variable especially the vectors should be considerable small. 
And for the case in the computation code, do we have any benchmark for the differences between the type in for-loops?